packages/aws-rfdk/lib/core/lambdas/nodejs/version-provider/handler.ts

```diff
@@ -8,51 +8,175 @@
+   */
+  readonly LinuxRepositoryInstaller: string;
+}
+
+/**
+ * This custom resource will parse and return the S3 URI's of the Deadline installers and Docker recipes for use in
+ * any constructs that need to install Deadline.
+ */
 export class VersionProviderResource extends SimpleCustomResource {
   readonly versionProvider: VersionProvider;
 
-  constructor(indexFilePath?: string) {
+  constructor() {
     super();
-    this.versionProvider = new VersionProvider(indexFilePath);
+    this.versionProvider = new VersionProvider();
   }
 
   /**
    * @inheritdoc
    */
-  /* istanbul ignore next */ // @ts-ignore
   public validateInput(data: object): boolean {
-    return this.versionProvider.implementsIVersionProviderProperties(data);
+    return this.implementsIVersionProviderResourceProperties(data);
   }
 
   /**
    * @inheritdoc
    */
   // @ts-ignore  -- we do not use the physicalId
-  public async doCreate(physicalId: string, resourceProperties: IVersionProviderProperties): Promise<Map<Platform, IVersionedUris>> {
-    return await this.versionProvider.getVersionUris(resourceProperties);
+  public async doCreate(physicalId: string, resourceProperties: IVersionProviderResourceProperties): Promise<FlatVersionedUriOutput> {
+    const deadlinePlatFormVersionedUris = await this.versionProvider.getVersionUris({
+      versionString: resourceProperties.versionString,
+      platform: Platform.linux,
+      product: Product.deadline,
+    });
+
+    const deadlineLinux = deadlinePlatFormVersionedUris.get(Platform.linux)!;
+    const deadlineLinuxUris = deadlineLinux.Uris;
+
+    const s3Bucket = this.parseS3BucketName(deadlineLinuxUris.bundle);
+    const linuxRepoObjectKey = this.parseS3ObjectKey(deadlineLinuxUris.repositoryInstaller!);
+
+    return {
+      S3Bucket: s3Bucket,
+      MajorVersion: deadlineLinux.MajorVersion,
+      MinorVersion: deadlineLinux.MinorVersion,
+      ReleaseVersion: deadlineLinux.ReleaseVersion,
+      LinuxPatchVersion: deadlineLinux.PatchVersion,
+      LinuxFullVersionString: Version.convertToFullVersionString(
+        deadlineLinux.MajorVersion,
+        deadlineLinux.MinorVersion,
+        deadlineLinux.ReleaseVersion,
+        deadlineLinux.PatchVersion,
+      ),
+      LinuxRepositoryInstaller: linuxRepoObjectKey,
+    };
   }
 
   /**
    * @inheritdoc
    */
   /* istanbul ignore next */ // @ts-ignore
-  public async doDelete(physicalId: string, resourceProperties: IVersionProviderProperties): Promise<void> {
+  public async doDelete(physicalId: string, resourceProperties: IVersionProviderResourceProperties): Promise<void> {
     // Nothing to do -- we don't modify anything.
     return;
   }
+
+  private implementsIVersionProviderResourceProperties(value: any): boolean {
+    if (!value || typeof(value) !== 'object') { return false; }
+
+    if (value.versionString) {
+      if (!Version.validateVersionString(value.versionString))  { return false; }
```

Comment:
It's probably worth adding a `console.log()` here if the validation of the version string fails just to provide some breadcrumbs in the lambda log.

If validation fails, then the customer will see "Input did not pass validation check" ( https://github.com/aws/aws-rfdk/blob/79e0d1f86389b289596974b15bcf49c735d72e56/packages/aws-rfdk/lib/core/lambdas/nodejs/lib/custom-resource/simple-resource.ts#L88 ) as the failure reason on the Custom Resource, but understanding why it failed the validation won't be obvious. Might actually be worth extending the error message in the base handler to point the customer to the Lambda log as well (the LambdaContext has the log group & stream name in it -- https://github.com/aws/aws-rfdk/blob/79e0d1f86389b289596974b15bcf49c735d72e56/packages/aws-rfdk/lib/core/lambdas/nodejs/lib/aws-lambda/types.ts#L13 ).

Reply:
Updated